test/unit/test_quil.py

```diff
@@ -1194,3 +1196,45 @@ def test_out_without_calibrations():
     combined_program = quilt_program + quil_program
 
     assert combined_program.out(calibrations=False) == quil_program.out()
+
+
+def test_program_equality():
+    program = Program("""DECLARE foo REAL[1]
+DEFFRAME 1 "rx":
+    HARDWARE-OBJECT: "hardware"
+DEFCAL I 0:
+    DELAY 0 1
+DEFCAL I 1:
+    DELAY 0 1
+DEFCAL I 2:
+    DELAY 0 1
+DEFCAL MEASURE 0 addr:
+    CAPTURE 0 "ro_rx" custom addr
+DEFCAL MEASURE 1 addr:
+    CAPTURE 1 "ro_rx" custom addr
+DEFWAVEFORM custom:
+    1,2
+DEFWAVEFORM custom2:
+    3,4
+DEFWAVEFORM another1:
+    4,5
+DEFGATE BAR AS MATRIX:
+    0, 1
+    1, 0
+DEFGATE FOO AS MATRIX:
+    0, 1
+    1, 0
+H 1
+CNOT 2 3""")
+
+    # Definitions in Quil are global in the sense that where they are defined in the program does not matter.
+    def is_global_state_instruction(i: AbstractInstruction) -> bool:
+        return isinstance(i, (DefFrame, DefCalibration, DefMeasureCalibration, DefWaveform, DefGate))
```

Comment:
This comment is incorrect; the order in which `DEFCAL` and `DEFCAL MEASURE` are defined _does_ matter. If the test passes with those reordered we've found a bug elsewhere.

Reply:
pyQuil, and now quil-rs, de-duplicate calibrations. If they didn't, then the statement would be false, right? I added a note to the comment to clarify. The spec is lacking on this subject, so let me know I'm missing something else.

Reply:
This has been fixed upstream, and this test has been updated to reflect respecting calibration ordering.